File: backend/app/schemas/nudge.py
"""Pydantic schemas for nudge API responses."""

from datetime import datetime

from pydantic import BaseModel, Field

from app.schemas.job import JobListingResponse


class SuggestedTodoResponse(BaseModel):
    id: str
    title: str
    due_date: datetime | None = None
    priority: int
    is_completed: bool


class NudgeResponse(BaseModel):
    """Response for the AI nudge endpoint."""

    headline: str
    bullets: list[str]
    type: str = Field(..., pattern="^(active|inactive)$")
    applications_this_week: int
    recommended_jobs: list[JobListingResponse]
    suggested_todos: list[SuggestedTodoResponse] = []

File: frontend/src/types/nudge.ts
import type { Job } from './job';

export interface SuggestedTodo {
  id: string;
  title: string;
  due_date: string | null;
  priority: number;
  is_completed: boolean;
}

export interface NudgeResponse {
  headline: string;
  bullets: string[];
  type: 'active' | 'inactive';
  applications_this_week: number;
  recommended_jobs: Job[];
  suggested_todos: SuggestedTodo[];
}
